### PR TITLE
Add filter: to_integer

### PR DESCRIPTION
**Pull request by crispgm**
Fix or support #5071

### CODE REVIEW

**Comment by crispgm** [comment]
Sorry for broke the checks.
I succeeded in passing all checks but failed on travis.

```
lib/jekyll/filters.rb:7:3: C: Metrics/ModuleLength: Module has too many lines. [242/240]
```
Can I get some help for fix error?

File: lib/jekyll/filters.rb [hunk 1]
```diff
@@ -260,6 +260,22 @@ def where_exp(input, variable, expression)
...
+    #
+    # Returns the integer value
+    def to_integer(input)
+      return input if input.is_a?(Integer)
+      return 1 if input == true
+      return 0 if input == false
+      begin
+        input.to_i
+      rescue
+        raise ArgumentError, "Invalid input object type."
```
Comment (parkr) [inline lib/jekyll/filters.rb]: This error should be more descriptive:

> Object '#{input.inspect}' could not be converted into an integer.

**Comment by parkr** [comment]
Above line 7 where it says `module Filters`, add `# rubocop:disable Metrics/ModuleLength`. :shrug:

File: lib/jekyll/filters.rb [hunk 2]
```diff
@@ -273,7 +273,7 @@ def to_integer(input)
       begin
         input.to_i
       rescue
-        raise ArgumentError, "Invalid input object type."
+        raise ArgumentError, "Object '#{input.inspect}' could not be converted into an integer."
```
Comment (parkr) [inline lib/jekyll/filters.rb]: To fix the `fmt` error, move this to two lines:

```ruby
raise ArgumentError, 
  "Object '#{input.inspect}' could not be converted into an integer."
```

Comment (crispgm) [inline lib/jekyll/filters.rb]: Thanks. It should be solved.
I learned the lesson that ```script/fmt``` should be run before commits. 😄

**Comment by parkr** [comment]
LGTM.

**Comment by parkr** [comment]
@jekyll/core: would someone else mind taking a look? Just need another LGTM.

File: lib/jekyll/filters.rb [hunk 3]
```diff
@@ -260,6 +261,23 @@ def where_exp(input, variable, expression)
...
+    #
+    # Returns the integer value
+    def to_integer(input)
+      return input if input.is_a?(Integer)
+      return 1 if input == true
+      return 0 if input == false
+      begin
+        input.to_i
+      rescue
+        raise ArgumentError,
```
Comment (envygeeks) [inline lib/jekyll/filters.rb]: IMO, this can just be removed, let Ruby's native error surface.

File: lib/jekyll/filters.rb [hunk 4]
```diff
@@ -260,6 +261,23 @@ def where_exp(input, variable, expression)
       end
     end
 
+    # Convert the input into integer
+    #
+    # input - the object string
+    #
+    # Returns the integer value
+    def to_integer(input)
+      return input if input.is_a?(Integer)
```
Comment (envygeeks) [inline lib/jekyll/filters.rb]: This is uncessary, Ruby is already optimized for this and just slows down to_i already.

**Comment by envygeeks** [comment]
Just a few comments! Thanks.

File: lib/jekyll/filters.rb (hunk 3 above)
Comment (crispgm) [inline lib/jekyll/filters.rb]: It's okay but if this was removed, users would get a ```undefined method `to_i'``` instead.
Will this error message mislead users?

Comment (envygeeks) [inline lib/jekyll/filters.rb]: There is nothing ambiguous about that error.  Our only job is to do what we are told to do, we can either be defensive or we can throw, if we choose to throw then the native error will suffice, they told us to run to_i, and to_i doesn't exist, therefore the native error is the most literally right, we need not rescue that as it was exactly what happened, if we choose to be defensive then return 0 like Ruby does with unpareseable strings.

Comment (crispgm) [inline lib/jekyll/filters.rb]: Thanks for your review and advice. The modification has been done. 😸

**Comment by envygeeks** [comment]
LGTM after a quick rebase!

**Comment by parkr** [comment]
LGTM, thank you.

**Comment by parkr** [comment]
LGTM.

**Comment by parkr** [comment]
2 LGTM's were given above. 👍 

@jekyllbot: merge +minor

**Pull request merged**